test/parser/schema_with_required_fields.py

```diff
@@ -0,0 +1,289 @@
+document_schema = {
+    "title": {
+        "type": "string",
+        "description": "The name of the document",
+        "required": True,
+    },
+    "document_url": {
+        "type": "url",
+        "actions": {"download": True},
+        "description": "A link to the document",
+        "required": True,
+    },
+}
+
+documents_schema = {
+    "documents": {
+        "type": "array",
+        "items": {
+            "type": "object",
+            "properties": {
+                "title": {
+                    "type": "string",
+                    "description": "The name of the document",
+                    "required": True,
+                },
+                "document_url": {
+                    "type": "url",
+                    "description": "A link to the document",
+                    "required": True,
+                },
+            },
+        },
+        "required": True,
+    }
+}
+
+contact_schema = {
+    "name": {
+        "type": "object",
+        "properties": {
+            "first_name": {
+                "type": "string",
+                "description": "The first name of the contact",
+                "required": True,
+            },
+            "last_name": {
+                "type": "string",
+                "description": "The last name of the contact",
+            },
```

Comment:
@asim-shrestha @mohamedmamdouh22 should we make fields required by default? Probably not eh?